### PR TITLE
[SDK] AMReX: update package.py 

### PR DESCRIPTION
**Pull request by mic84**
cc: @balay

### CODE REVIEW

File: var/spack/repos/builtin/packages/amrex/package.py [hunk 1]
```diff
@@ -26,61 +26,64 @@
...
 
-    homepage = "https://ccse.lbl.gov/AMReX/index.html"
+    homepage = "https://amrex-codes.github.io/amrex/"
     url      = "https://github.com/AMReX-Codes/amrex.git"
+
+    version('develop', git='https://github.com/AMReX-Codes/amrex.git',
+            branch='development')
 
-    version('17.06', git='https://github.com/AMReX-Codes/amrex.git', commit='836d3c7')
-    version('master', git='https://github.com/AMReX-Codes/amrex.git', tag='master')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/amrex/package.py]: Is there any reason to get rid of these other versions?

**Comment by mic84** [comment]
Yes, those versions do not work with the current cmake setup.
Michele

**Comment by balay** [comment]
@adamjstewart any idea what the CI failure means? [and how to fix the corresponding issue?]

**Comment by adamjstewart** [comment]
No idea, never seen that one before.

**Comment by balay** [comment]
Hm - perhaps this branch should be rebased/collapsed - so that the CI test is rerun?

**Comment by adamjstewart** [comment]
It's worth a shot.

**Comment by mbdevpl** [comment]
Hello! I'm moving the discussion from possibly obscure commit comments (https://github.com/spack/spack/commit/61aa1831e3d93a51abf38591d795e5507195c6c5) to here.

In my case even the 'develop' doesn't build, but when I quickly rewrote the AMReX Spack package to be  an `AutotoolsPackage` instead of `CmakePackage`, it builds without problems. Also the 'master' branch and some of the recent tags seem to be building using this approach. And then I'm running some actual scientific software that depends on AMReX, and it also seems to work...

I'm thinking about opening an issue and/or pull request, but first I'd like to discuss the topic here if that's ok.

The problem is:

1) Is using CmakePackage for AMReX somehow preferred over AutotoolsPackage?

2) Could 2 package versions (cmake & autotools) coexist somehow in Spack?

For now my modifications are in my fork: https://github.com/mbdevpl/spack/commit/56e0f23634a1f7b6a724571ba0e4d4924fa3435d

And error (experienced only in cmake) is here:
```
$ spack install amrex@develop dimensions=2 +eb +fortran ~openmp +particles +shared +mpi ^m
pich

...

[+] /work/mateusz/spack/opt/spack/linux-centos7-x86_64/gcc-8.2.0/python-2.7.15-q2wd2eokat6m7fiatdzu7gdb7rpwacxj
==> Installing amrex
==> Cloning git repository: https://github.com/AMReX-Codes/amrex.git on branch development
==> No checksum needed when fetching with git
==> Already staged amrex-develop-gr6wfbytka5vecetrue4ypek7zh7337a in /work/mateusz/spack/var/spack/stage/amrex-develop
-gr6wfbytka5vecetrue4ypek7zh7337a
==> No patches needed for amrex
==> Building amrex [CMakePackage]
==> Executing phase: 'cmake'
==> Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j20'

4 errors found in build log:
     838    CMakeFiles/amrex.dir/GeometryShop/AMReX_EllipsoidIF.cpp.o: In function `amrex::EllipsoidIF::EllipsoidIF(a
            mrex::EllipsoidIF const&)':
     839    AMReX_EllipsoidIF.cpp:(.text+0x240): multiple definition of `amrex::EllipsoidIF::EllipsoidIF(amrex::Ellip
            soidIF const&)'
     840    CMakeFiles/amrex.dir/GeometryShop/AMReX_Ellipsoid.cpp.o:AMReX_Ellipsoid.cpp:(.text+0x160): first defined
            here
     841    CMakeFiles/amrex.dir/GeometryShop/AMReX_EllipsoidIF.cpp.o: In function `amrex::EllipsoidIF::EllipsoidIF(a
            mrex::EllipsoidIF const&)':
     842    AMReX_EllipsoidIF.cpp:(.text+0x240): multiple definition of `amrex::EllipsoidIF::EllipsoidIF(amrex::Ellip
            soidIF const&)'
     843    CMakeFiles/amrex.dir/GeometryShop/AMReX_Ellipsoid.cpp.o:AMReX_Ellipsoid.cpp:(.text+0x160): first defined
            here
  >> 844    collect2: error: ld returned 1 exit status
  >> 845    make[2]: *** [Src/libamrex.so] Error 1
     846    make[2]: Leaving directory `/tmp/mateusz/spack-stage/spack-stage-CvlGag/amrex/spack-build'
  >> 847    make[1]: *** [Src/CMakeFiles/amrex.dir/all] Error 2
     848    make[1]: Leaving directory `/tmp/mateusz/spack-stage/spack-stage-CvlGag/amrex/spack-build'
  >> 849    make: *** [all] Error 2
```

For the record, I can reproduce the problem when I compile AMReX via cmake by hand, but only with certain options, e.g. `mkdir build && cd build && cmake .. -DENABLE_PARTICLES=ON -DBUILD_SHARED_LIBS=ON -DUSE_XSDK_DEFAULTS=ON -DENABLE_PIC=ON -DENABLE_EB=ON -DXSDK_ENABLE_Fortran=ON -DCMAKE_BUILD_TYPE=Release && make`.

**Comment by adamjstewart** [comment]
> 1. Is using CmakePackage for AMReX somehow preferred over AutotoolsPackage?

If both work, I have no personal preference, although others seems to prefer CMakePackage as it makes dependency location easier.

> 2. Could 2 package versions (cmake & autotools) coexist somehow in Spack?

Yes, see `arpack-ng`, `atk`, `dyninst`, `glog`, `metis`, `opennurbs`, and `superlu` for examples of packages that support mixed build systems.

**Comment by tgamblin** [comment]
That's a link problem -- I'm pretty sure it means something is wrong with the CMake build.  By default, Spack will add `-DCMAKE_BUILD_TYPE=RelWithDebInfo` to the command line, so this may be why the error is triggered.  This is a pretty common option for someone to add to your CMake build, so it would be good to make it work.

Does AMReX have two build systems?  If so, why?  I would pick the one that the developers actually use, and stick with that one.

**Comment by adamjstewart** [comment]
If `RelWithDebInfo` isn't supported, it's pretty easy to override that in the package.

**Comment by mic84** [comment]
RelWithDebInfo was not supported by our CMake system. AMReX is
xSDK-compatible, and, as such, it only supported Release and Debug build
types as per xSDK-guidelines.
That said, I just added support for all the CMake build types in the latest
commit on master branch: this should fix this issue with Spack.

Michele

**Comment by tgamblin** [comment]
@mic84: thanks! Can we get your consent over at #9137, as well?  We are attempting to relicense Spack to be more permissive (another xSDK recommendation).

**Comment by mic84** [comment]
My name is Michele Rosso, and I am a contributor to Spack. I have used or
may have used the following email address(es): mrosso@lbl.gov. I consent to
change the current license contained in the License file at
https://github.com/spack/spack/blob/develop/LICENSE from the LGPL-2.1
license to the MIT License and Apache License 2.0 with respect to all of my
contributions to Spack.

Michele

**Comment by tgamblin** [comment]
Thanks!

**Pull request merged**